commit 9e00f6ed1e84a3b251aded21ef98ed9e4adc6936
Author: sewardj <sewardj@a5019735-40e9-0310-863c-91ae7b9d1cf9>
Date:   Wed Nov 23 13:15:22 2016 +0000

    Bug 352767 - Wine/valgrind: Warning: noted but unhandled ioctl 0x5307
    with no size/direction hints. (CDROMSTOP).
    
    Patch from Austin English (austinenglish@gmail.com).
    
    
    git-svn-id: svn://svn.valgrind.org/valgrind/trunk@16152 a5019735-40e9-0310-863c-91ae7b9d1cf9

diff --git a/coregrind/m_syswrap/syswrap-linux.c b/coregrind/m_syswrap/syswrap-linux.c
index fda8dd1..5e540e8 100644
--- a/coregrind/m_syswrap/syswrap-linux.c
+++ b/coregrind/m_syswrap/syswrap-linux.c
@@ -5672,6 +5672,7 @@ PRE(sys_ioctl)
    
    /* CDROM stuff. */
    case VKI_CDROM_DISC_STATUS:
+   case VKI_CDROMSTOP:
 
    /* KVM ioctls that dont check for a numeric value as parameter */
    case VKI_KVM_S390_ENABLE_SIE:
@@ -9188,6 +9189,7 @@ POST(sys_ioctl)
 
       /* CD ROM stuff (??)  */
    case VKI_CDROM_DISC_STATUS:
+   case VKI_CDROMSTOP:
       break;
    case VKI_CDROMSUBCHNL:
       POST_MEM_WRITE(ARG3, sizeof(struct vki_cdrom_subchnl));
diff --git a/include/vki/vki-linux.h b/include/vki/vki-linux.h
index bdb8f33..7f6a117 100644
--- a/include/vki/vki-linux.h
+++ b/include/vki/vki-linux.h
@@ -1865,6 +1865,7 @@ struct vki_scsi_idlun {
                                 	           (struct cdrom_tochdr) */
 #define VKI_CDROMREADTOCENTRY		0x5306 /* Read TOC entry 
                                 	           (struct cdrom_tocentry) */
+#define VKI_CDROMSTOP			0x5307 /* Stop the cdrom drive */
 #define VKI_CDROMSUBCHNL		0x530b /* Read subchannel data 
                                 	           (struct cdrom_subchnl) */
 #define VKI_CDROMREADMODE2		0x530c /* Read CDROM mode 2 data (2336 Bytes) 

commit 88c8d843f64fcb56a0ddd492fb560e6594c3f2e6
Author: sewardj <sewardj@a5019735-40e9-0310-863c-91ae7b9d1cf9>
Date:   Wed Nov 23 13:26:23 2016 +0000

    Bug 348616 - Wine/valgrind: Warning: noted but unhandled ioctl 0x5390
    with no size/direction hints. (DVD_READ_STRUCT)
    
    Patch from Austin English (austinenglish@gmail.com).
    
    
    
    git-svn-id: svn://svn.valgrind.org/valgrind/trunk@16153 a5019735-40e9-0310-863c-91ae7b9d1cf9

diff --git a/coregrind/m_syswrap/syswrap-linux.c b/coregrind/m_syswrap/syswrap-linux.c
index 5e540e8..af10b92 100644
--- a/coregrind/m_syswrap/syswrap-linux.c
+++ b/coregrind/m_syswrap/syswrap-linux.c
@@ -5674,7 +5674,10 @@ PRE(sys_ioctl)
    case VKI_CDROM_DISC_STATUS:
    case VKI_CDROMSTOP:
 
-   /* KVM ioctls that dont check for a numeric value as parameter */
+   /* DVD stuff */
+   case VKI_DVD_READ_STRUCT:
+
+   /* KVM ioctls that don't check for a numeric value as parameter */
    case VKI_KVM_S390_ENABLE_SIE:
    case VKI_KVM_CREATE_IRQCHIP:
    case VKI_KVM_S390_INITIAL_RESET:
@@ -9226,6 +9229,10 @@ POST(sys_ioctl)
    case VKI_CDROM_GET_CAPABILITY: /* 0x5331 */
       break;
 
+      /* DVD stuff */
+   case VKI_DVD_READ_STRUCT:
+      break;
+
    case VKI_FIGETBSZ:
       POST_MEM_WRITE(ARG3, sizeof(unsigned long));
       break;
diff --git a/include/vki/vki-linux.h b/include/vki/vki-linux.h
index 7f6a117..b64b521 100644
--- a/include/vki/vki-linux.h
+++ b/include/vki/vki-linux.h
@@ -1885,6 +1885,7 @@ struct vki_scsi_idlun {
 #define VKI_CDROM_DISC_STATUS		0x5327	/* get CD type information */
 #define VKI_CDROM_GET_CAPABILITY	0x5331	/* get capabilities */
 
+#define VKI_DVD_READ_STRUCT		0x5390  /* read structure */
 #define VKI_CDROM_SEND_PACKET		0x5393	/* send a packet to the drive */
 
 struct vki_cdrom_msf0		
